{{cookiecutter.github_project_name}}/package.json

```diff
@@ -51,9 +51,10 @@
     "watch:nbextension": "webpack --watch"
   },
   "dependencies": {
-    "@jupyter-widgets/base": "^1.1.10 || ^2 || ^3"
+    "@jupyter-widgets/base": "^1.1.10 || ^2 || ^3 || ^4.0.0-beta.1"
```

Comment:
Wait, are we going to need a new major version of base? If so, maybe just going straight for 8.x is less painful?

Reply:
I won't be able to do an 8.0 release before jlab 3 ships, and I think we can't possibly expect people to do the more major changes in their packages needed for an 8.0 release before jlab 3 ships.

I think it's better to make it *very* easy to update to a jlab 3 compatible package right away, and make an upgrade path to update for the jlab 3 packaging, and sometime before the end of the year, make an 8.0 release.

Reply:
I imagine the easiest path of upgrades for a custom widget is to do these in order:

(a) bump the base version with an `||` if you want to preserve compatibility with previous versions. This is can be a patch release or minor release if you use `||`. This will require recompiling jlab, but will be very easy to update and release.
(b) Update to the jlab 3 bundle system using the cookiecutter or docs or update script as guides. This is a more major update, but hopefully has enough support to do relatively painlessly. This can be released as an update after jlab 3 comes out, and probably can be a minor or patch release.
(c) (after widgets 8.0 comes out) - update for widgets 8 - likely a major release.

Reply:
My experience from last time was that adding all the `||` introduces a large tail of unintended side effects (e.g. installs end up with duplicate versions of base installed). That might be less of an issue with lab 3 if we are successfully able to mark base as a singleton, but we might want to investigate if we risk damaging some lab2 installs.